zio-http/shared/src/main/scala/zio/http/Response.scala

```diff
@@ -58,6 +58,9 @@ final case class Response(
         self.copy(body = Body.fromChunk(bytes))
       }
 
+  def contentType(json: MediaType): Response =
+    self.addHeader(Header.ContentType(json))
```

Comment:
Rendering of the `ContentType` header is quite expensive. For this reason, at the bottom of the the `Response` companion object the main 3 content-types are cached after calling `untyped` on them (which invokes the `render` method).

Might be better to pattern match here and check if the media type here matches one of those 3 and use the cached value or fallback to `Header.ContentType.apply` if not?

Reply:
Also the param name should be changed to `mediaType`

Reply:
Maybe it is just the best to use the untyped API. Probably the cheapest.